arlas-server/src/main/java/io/arlas/server/app/DatabaseToolsFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Gisaïa under one or more contributor
+ * license agreements. See the NOTICE.txt file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Gisaïa licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.arlas.server.app;
+
+import com.codahale.metrics.health.HealthCheck;
+import io.arlas.server.dao.CollectionReferenceDao;
+import io.arlas.server.ogc.common.dao.OGCCollectionReferenceDao;
+import io.arlas.server.ogc.wfs.services.WFSToolService;
+import io.arlas.server.services.ExploreService;
+
+import java.util.Map;
+
+public abstract class DatabaseToolsFactory {
```

Comment:
Does the name match what it does? It looks to be more than tools

Reply:
Any other idea would be welcome!

Reply:
I guess it should be something like DatabaseServicesFactory, and dao should only be visible from a service-related class. Or we could use dependency injection. But as it is not like this today, I have no better idea neither.

Reply:
Yes, I'd like to go further into the refactoring, but I'm going slowly in order to limitate the risk and ease the PRs. We'll change the name later then.